runtime/src/main/java/org/corfudb/runtime/collections/PersistedStreamingMap.java

```diff
@@ -184,7 +187,7 @@ public V get(@NonNull Object key) {
     @Override
     public V put(@NonNull K key, @NonNull V value) {
         putCounter.ifPresent(Counter::increment);
-        Optional<Timer.Sample> recordSample = MicroMeterUtils.startTimer(putCounter, SAMPLING_RATE);
+        Optional<Timer.Sample> recordSample = MicroMeterUtils.startTimer(putCounter, randomNumberGenerator.nextInt(SAMPLE_SPACE));
```

Comment:
Maybe look if MicroMeter exposes some sampling functions ? seems like a common thing for a metrics library.

Reply:
It doesn't seem provide such functions.